build.gradle

```diff
@@ -125,10 +125,22 @@ processResources {
 }
 
 jar {
-
-    configurations.shade.each { dep ->
-        from(project.zipTree(dep)){
-            exclude 'META-INF', 'META-INF/**'
+    // Shading could try to add duplicate classes
+    // fail instead of silently adding multiple entries with the same name
+    duplicatesStrategy=DuplicatesStrategy.FAIL
```

Comment:
not `DuplicatesStrategry.EXCLUDE`?

Reply:
With these changes there aren't any duplicate entries, so duplicates should be a failure as if we did see duplicates it would indicate that something had gone wrong when shading.